test/location.cc

```diff
@@ -101,102 +101,112 @@ void test_from_json() {
   uint64_t way_id = 12987234107;
   std::string date_time = "2015-05-06T08:00";
 
-  try {
-    Location loc = Location::FromJson("something");
-    throw std::runtime_error("This should have been malformed json");
-  } catch (...) {
+  std::vector<Location::ParseMethod> methods{Location::ParseMethod::PTREE,
+    Location::ParseMethod::RAPIDJSON};
+
+  for(const auto& m : methods) {
+    try {
+      Location loc = Location::FromJson("something", m);
+      throw std::runtime_error("This should have been malformed json");
+    } catch (...) {
+    }
```

Comment:
the test fails if there is an uncaught exception. this bit of code is trying to make sure that bad json will throw an exception. to do that properly one should probably do something like:

```c++11
for(const auto& m : methods) {
  try {
    Location loc = Location::FromJson("something", m);
  } //this should catch something
  catch (...) {
    continue;
  }
  //if we got here then the above didnt throw and that is wrong
  throw std::runtime_error("This should have been malformed json");
}
```